jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -595,6 +595,21 @@ public static void readFrom(Path path, ByteBuffer buffer) throws IOException
         }
     }
 
+    public static void readFrom(ReadableByteChannel newReadableByteChannel, long contentLengthValue, ByteBuffer byteBuffer) throws IOException
+    {
+        int pos = BufferUtil.flipToFill(byteBuffer);
+        try (ReadableByteChannel channel = newReadableByteChannel)
+        {
+            int read = 0;
+            while (read != contentLengthValue)
```

Comment:
This loop might spin. Not only should it call `Thread.onSpinWait()` when it does (i.e.: when `read` returns 0) but that should also be clearly stated in the method's javadoc.

Reply:
I have updated this loop/javadoc also putting in a case for (read < 0) so please re-review.

Reply:
I have updated this loop/javadoc also putting in a case for (read < 0) so please re-review.